live_data_subscriber/__init__.py

```diff
@@ -65,10 +68,21 @@ def generate_topic_api_client_args_mapping():
             device_name = uuid_name_mapping[device_uuid]
             if device_name in allowed_devices_name_mapping:
                 topic_name = allowed_devices_name_mapping[device_name]
-                topic_api_client_mapping[topic_name].append({
-                    "device_id": device_uuid,
-                    **api_client_args
-                })
-
-    logging.info(f"Connecting to the following MQTT topics/devices {topic_api_client_mapping}")
-    return topic_api_client_mapping
+                topic_api_client_mapping[topic_name].append(
+                    {"device_id": device_uuid, **api_client_args}
+                )
+                device_api_client_mapping[device_name].append(
+                    {"device_id": device_uuid, **api_client_args}
+                )
+
+    logging.info(f"Connecting to the following MQTT topics {topic_api_client_mapping}")
+    logging.info(f"Connecting to the following WS devices {device_api_client_mapping}")
+    return topic_api_client_mapping, device_api_client_mapping
+
+
+def refresh_cn_and_device_list(last_time_checked, api_client_dict, is_mqtt=False):
```

Comment:
You have `api_client_dict` as input variable, but you never alter it. Is it really needed?

Reply:
Also, I dont see the need for returning `last_time_checked`. Were is it used outside of this function?

Reply:
I see, you pulled this out of the `MQTTConnection` class to use it also in `LiveDataWebsocketMessageReceiver`. 
IMO how it is implemented now it is not really pretty.I would vote for change by reference and omit the return values here. But this violates the guideline we agreed on (prefer return values over change by reference) and may also be premature. 
So, leave as is then